striatum/bandit/bandit.py

```diff
@@ -4,43 +4,60 @@
 from abc import abstractmethod
 
 
-# TODO: think about how to use this
 class Action(object):
-    """The action object"""
-    @abstractmethod
-    def __init__(self):
-        pass
+    """The action object
+
+        Parameters
+        ----------
+        action_id: int
+            The index of this action.
+
+        title: string
+            The title of this action.
+
+        content: object
+            The content of this action.
+    """
+    def __init__(self, action_id, title, content):
+        self.action_id = action_id
+        self.title = title
+        self.content = content
 
 
 class BaseBandit(object):
     """Bandit algorithm
 
-    Parameters
-    ----------
-    historystorage : historystorage object
-        The historystorage object to store history context, actions and rewards.
+        Parameters
+        ----------
+        historystorage : historystorage object
+            The historystorage object to store history context, actions and rewards.
+
+        modelstorage : modelstorage object
+            The modelstorage object to store model parameters.
 
-    modelstorage : modelstorage object
-        The modelstorage object to store model parameters.
+        actions : list of Action objects
+            List of actions to be chosen from.
 
-    actions : list of Action objects
-        List of actions to be chosen from.
+        Attributes
+        ----------
+        historystorage : historystorage object
+            The historystorage object to store history context, actions and rewards.
 
-    Attributes
-    ----------
-    historystorage : historystorage object
-        The historystorage object to store history context, actions and rewards.
+        modelstorage : modelstorage object
+            The modelstorage object to store model parameters.
 
-    modelstorage : modelstorage object
-        The modelstorage object to store model parameters.
+        actions : list of Action objects
+            List of actions to be chosen from.
 
-    actions : list of Action objects
-        List of actions to be chosen from.
+        actions_id: list of integers
+            List of all action_id's.
     """
+
     def __init__(self, historystorage, modelstorage, actions):
         self._historystorage = historystorage
         self._modelstorage = modelstorage
         self._actions = actions
+        self._actions_id = [actions[i].action_id for i in range(len(actions))]
```

Comment:
Instead of creating another variable, a possible design to extract what you want is to add a method.
```python
def get_action_ids(self):
    return [self._actions[i].action_id for i in range(len(self._actions))]
```